blazar_dashboard/content/leases/workflows.py

```diff
@@ -722,19 +745,55 @@ def handle(self, request, data):
                     return
 
                 lease = api.client.lease_get(self.request, lease_id)
-                fields['reservations'] = lease['reservations']
-                if len(fields['reservations']) != 1:
+                existing_reservations = [
+                    res for res in lease['reservations']
+                    if res["resource_type"] == "physical:host"
+                ]
+                if len(existing_reservations) != 1:
                     messages.error(request, "Cannot update node count for a lease "
-                                            "with multiple reservations.")
+                                            "with multiple host reservations.")
                     return
-                fields['reservations'][0]['min'] = min_hosts
-                fields['reservations'][0]['max'] = min_hosts
+                existing_reservations[0]['min'] = min_hosts
+                existing_reservations[0]['max'] = min_hosts
+                fields['reservations'] += existing_reservations
+                is_update = True
+
+        if conf.floatingip_reservation.get("enabled", False) and \
+                self.is_fipreservation_included:
+            amount_floatingips = data.get('amount_floatingips')
+            if amount_floatingips is not None:
+                try:
+                    amount_floatingips = int(amount_floatingips)
+                except ValueError as e:
+                    LOG.error('Error updating lease: %s', e)
+                    exceptions.handle(
+                        request, message="Invalid value provided.")
+                    return
+
+                lease = api.client.lease_get(self.request, lease_id)
+                existing_reservations = [
+                    res for res in lease['reservations']
+                    if res["resource_type"] == "virtual:floatingip"
+                ]
+                if len(existing_reservations) != 1:
+                    messages.error(request, "Cannot update node count for a lease "
+                                            "with multiple floating IP reservations.")
+                    return
+                existing_reservations[0]['amount'] = amount_floatingips
+                fields['reservations'] += existing_reservations
                 is_update = True
 
         reservations = data.get('reservations', None)
         if reservations:
-            fields['reservations'] = reservations
-            is_update = True
+            try:
+                fields['reservations'] += json.loads(reservations)
+                is_update = True
+            except json.JSONDecodeError:
+                exceptions.handle(request, message="Invalid JSON provided")
+                return
+
+        if len(fields['reservations']) == 0:
```

Comment:
Really just a style thing, but the "pythonic" way to do this, and I'm not even sure I 100% agree with it, is to simply say:

```python
if not fields['reservations']:
    # ...
```

It's less explicit, and so it can capture things like it being False or None, but in general that's fine for such business logic unless you really care about it being a list. Python tends to prefer code be more straightforward in such cases 🤷🏻 

This style of conditional is actually in PEP-8 https://www.python.org/dev/peps/pep-0008/

Reply:
I agree this is more straight-foward and pythonic, I believe it lingered from when I was testing various ways to implement this.